src/main/java/com/minecraftabnormals/upgrade_aquatic/common/entities/jellyfish/ColoredSizableJellyfishEntity.java

```diff
@@ -55,7 +61,18 @@ public void readAdditional(CompoundNBT compound) {
 	public ILivingEntityData onInitialSpawn(IServerWorld worldIn, DifficultyInstance difficultyIn, SpawnReason reason, ILivingEntityData spawnDataIn, CompoundNBT dataTag) {
 		spawnDataIn = super.onInitialSpawn(worldIn, difficultyIn, reason, spawnDataIn, dataTag);
 		boolean updateSize = false;
-		float size = this.getRNG().nextFloat() < 0.85F ? 1.0F : this.getRNG().nextBoolean() ? 0.5F : 0.65F;
+		float size = 0;
+		int denominator = 0;
+		for (int weight : naturalSpawningSizes.values())
+			denominator += weight;
```

Comment:
I'm not exactly recalculating the total here, it's supposed to be a way to select items randomly from a weighted pool. For example, with the default values the chance that the 0.5 size class will be selected is 3/40. If it is not selected, the chance for 0.65 is now 3/37. If that still isn't selected, the chance for the 1.0 size class is 34/34 and it will be used as the size. Instead using the total weight for all of these wouldn't be correct as it would treat already 'discarded' pool items as if they were still in, and have a chance to not select a size at all.

Reply:
Oh wait i didn't read you properly lmao, yes the total weight should be cached.